### DISCUSSION TITLE
ShortFileName attribute of Directory element in .wix file doesn't allow to use a localization variable

### DISCUSSION THREAD

**Discussion by MikeGratsas** [wixtoolset/issues#7922] [Questions]
According to WiX Toolset [documentation](https://wixtoolset.org/docs/schema/wxs/shortfilenametype/) ShortFileName attribute of Directory element should accept a localization variable with the format !(loc.VARIABLE). However, in opposite to WiX Toolset version 3, the current WiX Toolset version 4.0.3 allows to use constant values in 8.3 format only. Please answer, if the mentioned bug could be fixed?

**Comment by robmen** [comment]
What do you mean "the current WiX Toolset version 4.0.3 allows to use constant values in 8.3 format only."? Is there an error message somewhere?

Basically, you need to provide more detail to know where to suggest the problem is.

**Comment by MikeGratsas** [comment]
Exact error message compiling WiX project is:
`error WIX0026: The Directory/@ShortName attribute's value, '!(loc.ShortDirName)', is not a valid 8.3-compliant name. Legal names contain no more than 8 non-period characters followed by an optional period and extension of no more than 3 non-period characters. Any character except for the follow may be used: \ ? | > < : / * " + , ; = [ ] (space).`

**Reply by robmen** [reply]
That definitely looks like a bug. Feel free to [open an issue](https://github.com/wixtoolset/issues/issues/new?assignees=&labels=triage&projects=&template=bug_report.yml) for triage.
